softlayer/resource_softlayer_global_ip.go

```diff
@@ -45,6 +47,16 @@ func resourceSoftLayerGlobalIp() *schema.Resource {
 			"routes_to": &schema.Schema{
 				Type:     schema.TypeString,
 				Required: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					address := v.(string)
+					if net.ParseIP(address) == nil {
+						errors = append(errors, fmt.Errorf("Invalid IP"))
+					}
+					return
+				},
+				DiffSuppressFunc: func(k, o, n string, d *schema.ResourceData) bool {
+					return net.ParseIP(o).String() == net.ParseIP(n).String()
+				},
 			},
```

Comment:
In `DiffSuppressFunc`, the value `o` will have full ipv6 address format. However, `n` can have both full format or compressed format. So before compare `o` and `n`, it is necessary to uncompress `n`.